src/components/__tests__/SearchBar.spec.ts

```diff
@@ -0,0 +1,167 @@
+                    style: SearchBar.parseStyle("{{...}}")
+                },
+                createElement("div", { className: "search-container" },
+                    createElement("span", { className: "glyphicon glyphicon-search" }),
+                    createElement("input", { className: "form-control", placeholder: "" }),
+                    createElement("button", { className: "btn-transparent" },
+                        createElement("span", { className: "glyphicon glyphicon-remove" })
+                    )
+                )
+            )
+        );
+    });
+
+    it("is not visible when show appearance is set to no", () => {
+        const searchBarProps: SearchBarProps = {
+            defaultQuery: "search",
+            onTextChangeAction:  jasmine.any(Function) as any,
+            placeHolder: "",
+            showSearchBar: false,
+            style: SearchBar.parseStyle("html{}")
+        };
+        const searchBar = renderSearchBar(searchBarProps);
+
+        expect(searchBar).toBeElement("");
+    });
+
+    it("shows place holder when specified", () => {
+        const searchBarProps: SearchBarProps = {
+            defaultQuery: "",
+            onTextChangeAction:  jasmine.any(Function) as any,
+            placeHolder: "Search",
+            showSearchBar: true,
+            style: SearchBar.parseStyle("html{}")
+        };
+        const searchBar = renderSearchBar(searchBarProps);
+
+        expect(searchBar).toBeElement(
+            createElement("div",
+                {
+                    className: "widget-offline-search",
+                    style: SearchBar.parseStyle("{{...}}")
+                },
+                createElement("div", { className: "search-container" },
+                    createElement("span", { className: "glyphicon glyphicon-search" }),
+                    createElement("input", { className: "form-control", placeholder: "Search" }),
+                    createElement("button", { className: "btn-transparent" },
+                        createElement("span", { className: "glyphicon glyphicon-remove" })
+                    )
+                )
+            )
+        );
+    });
+
+    describe("input", () => {
+        it("accepts value", () => {
+            const searchBarProps: SearchBarProps = {
+                defaultQuery: "search bar",
+                onTextChangeAction:  () => { return; },
+                placeHolder: "",
+                showSearchBar: true,
+                style: SearchBar.parseStyle("html{}")
+            };
+
+            const searchBarComponent = TestUtils.renderIntoDocument(createElement(SearchBar, searchBarProps));
+            const inputField = TestUtils.findRenderedDOMComponentWithTag(searchBarComponent, "input");
+            TestUtils.Simulate.change(inputField);
+
+            expect((inputField as HTMLInputElement).value).toBe("search bar");
+        });
+
+        it("has input when default query is specified", () => {
```

Comment:
Its different...first one similates onChange whereas this one tests if the input has a value at the time of render....
the wording should be `renders with specified default query`